fixing misalignment for right justification

Sega-Zero opened the pull request:
same as #1 but for right justification

eroth commented:
Hmm, have you seen this behavior under your fix? This is on my example project with the `ERCollectionViewCell` width set to `200` in the xib.  This is somewhat similar to the issue I was encountering in Issue #4, although it's happening with more frequency under `FlowLayoutHorizontalJustificationRight`.

![ios simulator screen shot aug 13 2015 8 34 47 am](https://cloud.githubusercontent.com/assets/1851313/9249806/2e980804-4196-11e5-8224-26f28870f3ad.png)

Sega-Zero committed:
more misalignment fix

fix the case when horizontalJustification ==
FlowLayoutHorizontalJustificationLeft and the next item may be fully
fit in the same line

Sega-Zero committed:
Merge remote-tracking branch 'origin/master'

Sega-Zero committed:
another approach in fixing misalignments
